doc/sphinx/language/core/variants.rst

```diff
@@ -24,6 +28,33 @@ Variants
    .. exn:: The @natural th argument of @ident must be @ident in @type.
       :undocumented:
 
+.. example::
+
+  The Booleans, the unit type and the empty types are respectively defined by:
+
+   .. coqtop:: in
+
+      Variant bool : Type := true : bool | false : bool.
+      Variant unit : Type := tt : unit.
+      Variant Empty_set : Type :=.
+
+  The option and sum types are defined by:
+
+   .. coqtop:: in
+
+      Variant option (A : Type) : Type := None : option A | Some : A -> option A.
+      Variant sum (A B : Type) : Type := inl : A -> sum A B | inr : B -> sum A B.
+
+  The Boolean reflection and equality predicates are defined by:
+
+  .. coqtop:: in
+
+     Variant reflect (P : Prop) : bool -> Set :=
+     | ReflectT : P -> reflect P true
+     | ReflectF : ~ P -> reflect P false.
+     Variant eq (A : Type) (x : A) : A -> Prop :=  eq_refl : eq x x.
```

Comment:
Do we already know about implicit argument at this point? Otherwise maybe
```suggestion
     Variant eq (A : Type) (x : A) : A -> Prop :=  eq_refl : @eq A x x.
```
could help?
Would those two type require a bit more explanation?

Reply:
That change still needs knowledge of implicit arguments to explan the @ 
Maybe disable automatic implicits instead